src/terrain/terrain.js

```diff
@@ -240,7 +243,13 @@ export class Terrain extends Elevation {
         this._tilesDirty = {};
         this.style = style;
         this._useVertexMorphing = true;
+    }
+
+    set style(style: Style) {
         style.on('data', this._onStyleDataEvent.bind(this));
+        style.on('neworder', this._checkRenderCacheEfficiency.bind(this));
```

Comment:
Do we really need this `neworder` event? I think its fine to not have it

Reply:
I would also prefer without, but this event is only necessary when order changes _after_ setting the style in terrain, I'll look into ways to bypass that but we might have to keep this one.